io-sim/src/Control/Monad/IOSimPOR/Internal.hs

```diff
@@ -814,6 +816,11 @@ schedule thread@Thread{
       let thread' = thread { threadControl = ThreadControl k ctl }
       schedule thread' simstate
 
+    NewUnique k -> do
+      let thread'   = thread{ threadControl = ThreadControl (k nextUniq) ctl }
+          simstate' = simstate{ nextUniq = MkUnique (unMkUnique nextUniq + 1) }
+      schedule thread' simstate'
```

Comment:
We should add a trace event here for creating a new unique.

Reply:
Done.